pkg/operator/ceph/cluster/cluster.go

```diff
@@ -181,5 +181,19 @@ func clusterChanged(oldCluster, newCluster cephv1beta1.ClusterSpec) bool {
 		changeFound = true
 	}
 
+	if oldCluster.Mon.Count != newCluster.Mon.Count {
+		logger.Infof("number of mons have changed from %d to %d. The health check will update the mons...", oldCluster.Mon.Count, newCluster.Mon.Count)
+		clusterRef.mons.MonCountMutex.Lock()
+		clusterRef.mons.Count = newCluster.Mon.Count
+		clusterRef.mons.MonCountMutex.Unlock()
+	}
+
+	if oldCluster.Mon.AllowMultiplePerNode != newCluster.Mon.AllowMultiplePerNode {
+		logger.Infof("allow multiple mons per node changed from %t to %t. The health check will update the mons...", oldCluster.Mon.AllowMultiplePerNode, newCluster.Mon.AllowMultiplePerNode)
+		clusterRef.mons.MonCountMutex.Lock()
```

Comment:
How about moving this mutex to `clusterRef`? `mon.count` will definitely not be the only option that is/will be update-able now and/or in the future that will need a lock.